### DISCUSSION TITLE
Shopware 6 CMS Plugins extend checkout

### DISCUSSION THREAD

**Discussion by meeshoogendoorn** [vuestorefront/shopware-pwa#1829] [General]
Hello, We are using a customer loyalty plugin and i created a shopware pwa plugin in Shopware 6 Plugin for this loyalty system. 

See the following image for a preview:
<img width="497" alt="Screenshot 2022-02-24 at 13 42 14" src="https://user-images.githubusercontent.com/55089200/155526867-4e48b997-d8cc-4d07-b0c9-2dc54b603981.png">

The checkbox can be selected by the customer when they want to get some credits to get free offers.
This plugin is injected from Shopware 6 using `npx @shopware-pwa/cli plugins`.

What is the best way to get the result of the checkbox if it is selected or not and keep it plug'n play?

The Shopware 6 plugin requires to send a parameter `piggyCollect: true` with the invokeCreateOrder request.

**Comment by niklaswolf** [comment]
That's a very interesting question! Here are my thoughts:
I think there is currently no way to do this on the shopware-pwa side directly.
But maybe it could be possible to do it like this (just an idea):

1. User changes checkbox, checkbox has a change-handler which triggers a custom api-call which adds a context extension(maybe this could be also done with an extension of the Context? So maybe the context can be patched directly)
2. The shopware-plugin adds a event listener on the order-created-event and does whatever it has to do depending on the value of the context extension

Like this it would be plug'n'play on the PWA. :)

**Reply by meeshoogendoorn** [reply]
Well i currently did a fix with axios interceptors when calling the place-order route it will inject a parameter to the body. I only don't like this solution. I would like to somehow extend this like we can in Shopware 6 so we can make checkout Plugins plug'n and play for Shopware PWA. I will look into the context Extension to check if this will be a sustainable soltion

**Reply by niklaswolf** [reply]
The "cleanest" solution would probably be to add a custom entity which saves the relation from context to whatever has to be persisted and adding a custom store-api endpoint to alter the values. Like this the context itself doesn't get cluttered. 
The values can then be fetched later in the order-created-event by shopware.

**Comment by sgc-fireball** [comment]
Hey,
we have nearly the same issue / request for the registration formular with "yes i wan't the newsletter".

EDIT:
I think it is a possible solution, that the "send" action in all cases collect the dom form params.
